tests/acl/test_acl.py

```diff
@@ -3,796 +3,768 @@
             with loganalyzer:
-                self.setup_rules(duthost, setup, acl_table)
-            self.post_setup_hook(duthost, localhost)
+                self.setup_rules(duthost, acl_table)
+
+            self.post_setup_hook(duthost, localhost, populate_vlan_arp_entries)
         except LogAnalyzerError as err:
-            # cleanup config DB in case of log analysis error
-            self.teardown_rules(duthost, setup)
+            # Cleanup Config DB if rule creation failed
+            logger.error("ACL table creation failed, attempting to clean-up...")
+            self.teardown_rules(duthost)
             raise err
 
         try:
             yield
         finally:
             loganalyzer.expect_regex = [LOG_EXPECT_ACL_RULE_REMOVE_RE]
             with loganalyzer:
-                self.teardown_rules(duthost, setup)
+                logger.info("Removing ACL rules")
+                self.teardown_rules(duthost)
 
-    @pytest.yield_fixture(scope='class', autouse=True)
+    @pytest.yield_fixture(scope="class", autouse=True)
     def counters_sanity_check(self, duthost, acl_rules, acl_table):
+        """Validate that the counters for each rule in the rules list increased as expected.
+
+        This fixture yields a list of rule IDs. The test case should add on to this list if
+        it is required to check the rule for increased counters.
+
+        After the test cases pass, the fixture will wait for the ACL counters to update and then
+        check if the counters for each rule in the list were increased.
+
+        Args:
+            duthost: The DUT having ACLs applied.
+            acl_rules: Fixture that sets up the ACL rules.
+            acl_table: Fixture that sets up the ACL table.
+
         """
-        counters sanity check after traffic test cases.
-        This fixture yields python list of rule IDs which test case should extend if
-        the RULE is required to check for increased counters.
-        After test cases passed the fixture will wait for ACL counters to update
-        and check if counters for each rule in the list of rules were increased.
-        :param duthost: DUT host object
-        :param acl_rules: rules creating fixture
-        :param acl_table: table creating fixture
-        :return:
-        """
+        table_name = acl_table["table_name"]
+        acl_facts_before_traffic = duthost.acl_facts()["ansible_facts"]["ansible_acl_facts"][table_name]["rules"]
 
-        table_name = acl_table['name']
-        acl_facts_before_traffic = duthost.acl_facts()['ansible_facts']['ansible_acl_facts'][table_name]['rules']
         rule_list = []
         yield rule_list
 
         if not rule_list:
             return
 
-        # wait for orchagent to update ACL counters
-        time.sleep(self.ACL_COUNTERS_UPDATE_INTERVAL)
+        # Wait for orchagent to update the ACL counters
+        time.sleep(self.ACL_COUNTERS_UPDATE_INTERVAL_SECS)
```

Comment:
Is there any chance to wait for ASIC db?

Reply:
It's possible, but I think the static wait is OK here since it is depending on the ACL counter update interval, which is a static value. We could improve upon this by maybe fetching the polling interval from the DB, but this is probably better left for a future PR.